tests/unit/s2n_psk_test.c

```diff
@@ -574,174 +612,391 @@ int main(int argc, char **argv)
 
             struct s2n_psk *psk = NULL;
             EXPECT_OK(s2n_array_pushback(&conn->psk_params.psk_list, (void**) &psk));
-            EXPECT_SUCCESS(s2n_psk_init(psk, S2N_PSK_TYPE_RESUMPTION));
-            EXPECT_SUCCESS(s2n_psk_new_identity(psk, identity.data, identity.size));
-            EXPECT_SUCCESS(s2n_psk_new_secret(psk, resumption_secret.data, resumption_secret.size));
+            EXPECT_OK(s2n_psk_init(psk, S2N_PSK_TYPE_RESUMPTION));
+            EXPECT_SUCCESS(s2n_psk_set_identity(psk, identity.data, identity.size));
+            EXPECT_SUCCESS(s2n_psk_set_secret(psk, resumption_secret.data, resumption_secret.size));
 
             EXPECT_OK(s2n_finish_psk_extension(conn));
             EXPECT_EQUAL(s2n_stuffer_data_available(&conn->handshake.io), full_client_hello.size);
             EXPECT_BYTEARRAY_EQUAL(conn->handshake.io.blob.data, full_client_hello.data, full_client_hello.size);
 
             EXPECT_SUCCESS(s2n_connection_free(conn));
         }
+    }
+
+    /* Test s2n_connection_set_external_psks */
+    {
+        uint8_t identity_0[] = "identity";
+        uint8_t secret_0[] = "secret";
 
-        /* Test s2n_connection_set_external_psks */
+        uint8_t identity_1[] = "identity 1";
+        uint8_t secret_1[] = "secret 1";
+
+        uint8_t huge_identity[UINT16_MAX] = { 0 };
+
+        DEFER_CLEANUP(struct s2n_psk *first_psk = s2n_external_psk_new(), s2n_psk_free);
+        EXPECT_SUCCESS(s2n_psk_set_identity(first_psk, identity_0, sizeof(identity_0)));
+        EXPECT_SUCCESS(s2n_psk_set_secret(first_psk, secret_0, sizeof(secret_0)));
+        EXPECT_SUCCESS(s2n_psk_set_hmac(first_psk, S2N_PSK_HMAC_SHA384));
+
+        DEFER_CLEANUP(struct s2n_psk *second_psk = s2n_external_psk_new(), s2n_psk_free);
+        EXPECT_SUCCESS(s2n_psk_set_identity(second_psk, identity_1, sizeof(identity_1)));
+        EXPECT_SUCCESS(s2n_psk_set_secret(second_psk, secret_1, sizeof(secret_1)));
+        EXPECT_SUCCESS(s2n_psk_set_hmac(second_psk, S2N_PSK_HMAC_SHA384));
```

Comment:
Might be worth adding session resumption PSK to the list here as `s2n_connection_set_external_psks` includes different logic for  `if (existing_psk->type == S2N_PSK_TYPE_EXTERNAL) ` or not on line tls/s2n_psk.c:329.

Reply:
I'm not sure I understand? These PSKs are just created to be used as common arguments for later test calls to s2n_connection_set_external_psks. They're not being added to any list, and a resumption PSK wouldn't be valid.

If you're asking for a test that verifies behavior when the connection already has several resumption PSKs, see "Ensures existing external psks are deleted and existing resumption psks are not deleted"